Request OPTIONS fix to return all allowed methods

michalbundyra opened the pull request:
The integration tests are pulled from zend-expressive-router repository.

See: https://github.com/zendframework/zend-expressive-router/pull/53

michalbundyra committed:
Fixed broken tests

The interface of the latest zend-expressive-router has been changed.
We can't get matched middleware from RouteResult but we can get matched
route, and then from route we can get middleware.

michalbundyra committed:
Added integration tests for HEAD and OPTIONS requests

michalbundyra committed:
Fixed HEAD and OPTIONS requests

michalbundyra committed:
Updated integration tests - explicit HEAD and OPTIONS requests

michalbundyra committed:
Use constant instead of string - RequestMethod

michalbundyra committed:
Updated ZendRouter to work with explicit and implicit HEAD and OPTIONS requests

michalbundyra committed:
Removed redundant condition

geerteltink commented on composer.json:
```diff
@@ -19,17 +19,23 @@
+    "repositories": [
+        {
+            "type": "git",
+            "url": "https://github.com/webimpress/zend-expressive-router"
```
I guess this is here for testing only as it should never be merged.

michalbundyra replied:
This PR depends on  zendframework/zend-expressive-router#53 and there is no other way (or I don't know it) to test it than including the repository from my branch with changes. Once  zendframework/zend-expressive-router#53 is merged I'll update this (and also other PRs).

geerteltink commented on composer.json:
```diff
@@ -19,17 +19,23 @@
         "fig/http-message-util": "^1.1.2",
         "psr/http-message": "^1.0.1",
-        "zendframework/zend-expressive-router": "^3.0.0alpha1 || ^3.0",
+        "zendframework/zend-expressive-router": "dev-hotfix/options-request",
```
Same, just a reminder to change this before merging.

geerteltink commented on composer.json:
```diff
@@ -19,17 +19,23 @@
     "require-dev": {
         "malukenho/docheader": "^0.1.6",
-        "phpunit/phpunit": "^6.5.3",
+        "phpunit/phpunit": "^7.0.2",
```
Can we please stop with all these bugfix requirements all the time? Why is 7.0.2 needed? Do the tests fail with ^7.0?

michalbundyra replied:
For all packages we are always using bugfixes version, why we shouldn't do the same for PHPUnit? It's the latest version available, and it doesn't matter too much as it is just testing framework. I wouldn't bother about it at all.

weierophinney replied:
I'm with @xtreamwayz here; if you're not updating the requirements for a specific reason, there's no need to bump them typically.

(That said, I won't block this PR based on that; just noting that we shouldn't be bumping requirements on every patch, but only when we really need to.)

michalbundyra committed:
HEAD request match GET request if there is no explicit HEAD route match

michalbundyra committed:
Updates to zend-expressive-router 3.0.0rc1

Pins to 3.0.0rc1, and updates new test case based on name change in that
release.

weierophinney commented on test/IntegrationTest.php:
```diff
@@ -0,0 +1,22 @@
+namespace ZendTest\Expressive\Router;
+
+use Zend\Expressive\Router\RouterInterface;
+use Zend\Expressive\Router\Test\IntegrationTest as RouterIntegrationTest;
```
I renamed the test case when I merged the patch on zend-expressive-router; it's now `ImplicitMethodsIntegrationTest`.

michalbundyra committed:
Updated integration test class name
